src/pipelinedefinition.cpp

```diff
@@ -63,6 +63,12 @@ Status PipelineDefinition::validate(ModelManager& manager) {
     if (!validationResult.ok()) {
         return validationResult;
     }
+
+    validationResult = validateDemultiplexerGatherNodesOrder();
```

Comment:
There are plenty of tests which manually call `validateNodes+validateForCycles` in order to check if pipeline is validated. Since this change will introduce third function - `validateDemultiplexerGatherNodesOrder`, we might have alot of tests passing and not using `validateDemultiplexerGatherNodesOrder`. Options are - move this logic to either `validateNodes` or `validateForCycles` or fix tests. First option will most likely be better option.

Reply:
After checking other parts of this MR I think it would be better to have pipeline validated for cycles first. Thats why option 2 is most likely better option.